core/src/main/java/org/apache/calcite/adapter/enumerable/RexToLixTranslator.java

```diff
@@ -727,11 +736,38 @@ private Expression translateParameter(RexDynamicParam expr,
     if (storageType == null) {
       storageType = typeFactory.getJavaClass(expr.getType());
     }
-    return nullAs.handle(
-        convert(
-            Expressions.call(root, BuiltInMethod.DATA_CONTEXT_GET.method,
-                Expressions.constant("?" + expr.getIndex())),
-            storageType));
+
+    Expression expression = Expressions.call(root, BuiltInMethod.DATA_CONTEXT_GET.method,
+            Expressions.constant("?" + expr.getIndex()));
+
+    Set<SqlTypeName> toIntegerTypes = Sets.newHashSet(
+            DATE,
+            TIME,
+            TIME_WITH_LOCAL_TIME_ZONE,
+            INTERVAL_YEAR,
+            INTERVAL_YEAR_MONTH,
```

Comment:
If we only support `DATE` and `TIME` type, just enum the 2 is enough, or better, we do not need this enum Set but add another convert phrase for the initial expression,

Reply:
Ok， i will remove the type that is not supported now